### PR TITLE
added ability to find material names in DAG

### PR DESCRIPTION
**Pull request by shimwell**
As discussed in issue #2229 this PR is an attempt to allow users to easily find the DAGMC material tags present in a DAGMCUniverse.

This is useful as the tags are needed when naming openmc.Material() objects used in the simulation and there is not an easy way of finding tags from a material file otherwise. openmc.Materials with the same names as the tags must exist for each tag otherwise the simulation fails.

There is this [handy micro package](https://github.com/fusion-energy/dagmc_h5m_file_inspector) I made a while back that uses MOAB to find tag names but no one wants to have to install another package for such a task and it would be much more convenient to have the material tags where they are needed. 

Also the method is proposed in the PR is nicer than the dagmc_h5m_file_inspector package as it makes use of h5py which is a openmc dependency already.

This PR is moving functionality from another one of my mico packages and improving it along the way.

### CODE REVIEW

File: openmc/universe.py
Comment (pshriwise) [inline openmc/universe.py:723]:
```suggestion
            candidate_tag = tag.tobytes().decode().replace('\x00', '')
```

File: openmc/universe.py
Comment (pshriwise) [inline openmc/universe.py:730]: Why do these need to go into a set first?

File: openmc/universe.py
Comment (pshriwise) [inline openmc/universe.py:711]:
```suggestion
        materials : list of str
```

**Comment by pshriwise** [comment]
I like that these DAGMC universes are gaining more of the capabilities our other geometry classes have! It would be nice to somehow name this method `get_all_materials` like other classes have, but I don't know how we'd return a similar data structure here as we don't have any `openmc.Material` objects to return.

**Review by pshriwise — CHANGES_REQUESTED** [review]
Didn't notice that this was a draft at first, but here's a pre-emptive review @shimwell. Just a few thoughts on how to simplify the string conversion.

File: openmc/universe.py
Comment (shimwell) [reply openmc/universe.py:730]: My thinking here was a set would remove any duplicate material tags. [m1,m1m1,m2] would become [m1,m2]

File: openmc/universe.py
Comment (shimwell) [reply openmc/universe.py:723]: That is much nicer, thanks Patrick

**Comment by shimwell** [comment]
Thanks for the review @pshriwise , I just keep them as draft till I've seen that they pass the tests. Many thanks for improving the hex to ascii conversion part

File: openmc/universe.py
Comment (pshriwise) [reply openmc/universe.py:727]: Ah I see. Yeah I guess that multiple groups _can_ be created for the same material.

File: openmc/universe.py
Comment (paulromano) [inline openmc/universe.py:704]: For `@property`s, you need to add the description in the "Attributes" section of the class docstring for it to show up in the documentation.

Comment (shimwell) [reply openmc/universe.py]: Sorry about that, I've moved them as suggested 👍

**Comment by shimwell** [comment]
Just putting this PR on hold as we have a regular DAGMC meeting on Thursday and wanted to invite the DAGMC community to comment

**Comment by shimwell** [comment]
Discussed at the recent DAGMC meeting https://github.com/svalinn/DAGMC/discussions/833#discussion-4432125 ✔️

**Review by pshriwise — APPROVED** [review]

**Comment by pshriwise** [comment]
I'll merge tomorrow morning unless @paulromano has additional comments.

**Review by paulromano — APPROVED** [review]
LGTM

**Pull request merged**
